### DISCUSSION TITLE
Multi clusters management

### DISCUSSION THREAD

**Discussion by tuatara73** [argoproj/argo-cd#11306] [Q&A]
Hi, could someone point me/answer here about how to manage correctly multi clusters in argocd?
The main question is should I use different name for apps for every cluster, e.g. cluster1-cert-manager and cluster2-cert-manager.
Or is there a better approach?
Couldn't find anything of value for this question in multiple websites.

**Comment by marcelfranca** [comment]
Your example is correct, if you have multiple apps with the same name ArgoCD will behave erratically.

**Comment by nastacio** [comment]
I would generalize apps so that environment-specific settings are set as parameters (there are different parameterization/templating choices, such as Helm and Kustomize) and the actual values of those environment-specific settings are isolated in environment-specific folders or files. 

I wrote a couple of blog entries on this topic and I think "#3" in ["Design strategies for GitOps repositories"](https://dnastacio.medium.com/gitops-repositories-the-right-way-part-1-mapping-strategies-6409dff758b5) addresses your question directly.
